File: problems/LeetCode/TopInterview150/jump.ts
function jump(nums: number[]): number {
  let pos = nums.length - 1;
  let steps = 0;

  while (pos !== 0) {
    let farthest = pos;

    for (let i = pos - 1; i >= 0; i--) {
      if (nums[i] >= pos - i) {
        farthest = i;
      }
    }

    pos = farthest;
    steps++;
  }

  return steps;
}

File: problems/LeetCode/TopInterview150/jump.py
class Solution:
    def jump(self, nums: List[int]) -> int:

        n = len(nums)
        goal = n - 1
        steps = 0

        while goal != 0:
            furthest = goal
            for i in range(goal - 1, -1, -1):
                if i + nums[i] >= goal:
                    furthest = i
            goal = furthest
            steps += 1

        return steps
